libs/math/include/math/mat4.h

```diff
@@ -319,6 +319,79 @@ class MATH_EMPTY_BASES TMat44 :
         return TMat33<T>(m_value[0].xyz, m_value[1].xyz, m_value[2].xyz);
     }
 
+    /**
+     * Decomposes a matrix into "TRS" form, this can be useful when reading certain asset file
+     * formats. Assumes no skew or shear.
+     */
```

Comment:
The comment should talk about how to use the TRS to recreate the matrix, i.e. is it:

`m = t*r*s;`
or
`m = s*r*t;`

Also I think this should return the 3x3 `R` matrix instead of the quaternion, callers can easily call `toQuaternion()` on their own.